javav2/example_code/opensearch/src/main/java/com/example/search/scenario/OpenSearchScenario.java

```diff
@@ -0,0 +1,323 @@
+        while (true) {
+            logger.info("");
+            logger.info("Enter 'c' followed by <ENTER> to continue:");
+            String input = scanner.nextLine();
+
+            if (input.trim().equalsIgnoreCase("c")) {
+                logger.info("Continuing with the program...");
+                logger.info("");
+                break;
+            } else {
+                logger.info("Invalid input. Please try again.");
+            }
+        }
+    }
+
+    private static void runScenario() throws Throwable {
+        String currentTimestamp = String.valueOf(System.currentTimeMillis());
+        String domainName = "test-domain-" + currentTimestamp;
+
+        logger.info(DASHES);
+        logger.info("1. Create an Amazon OpenSearch domain");
+        logger.info("""
+            An Amazon OpenSearch domain is a managed instance of the OpenSearch engine, 
+            which is an open-source search and analytics engine derived from Elasticsearch. 
+            An OpenSearch domain is essentially a cluster of compute resources and storage that hosts 
+            one or more OpenSearch indexes, enabling you to perform full-text searches, data analysis, and 
+            visualizations.
+
+            In this step, we'll initiate the creation of the domain. We'll check on the progress in a later step.
+        """);
+        waitForInputToContinue(scanner);
+
+        try {
+            CompletableFuture<String> future = openSearchActions.createNewDomainAsync(domainName);
+            String domainId = future.join();
+            logger.info("Domain successfully created with ID: {}", domainId);
+        } catch (RuntimeException rt) {
+            Throwable cause = rt.getCause();
+            if (cause != null) {
+                if (cause instanceof OpenSearchException openSearchEx) {
+                    logger.error("OpenSearch error occurred: Error message: {}, Error code {}", openSearchEx.awsErrorDetails().errorMessage(), openSearchEx.awsErrorDetails().errorCode());
+                } else {
+                    logger.error("An unexpected error occurred: " + cause.getMessage(), cause);
+                }
+            } else {
+                logger.error("An unexpected error occurred: " + rt.getMessage());
+            }
+            throw cause;
+        }
+        waitForInputToContinue(scanner);
+        logger.info(DASHES);
+
+        logger.info("2. Describe the Amazon OpenSearch domain");
+        logger.info("In this step, we get back the Domain ARN which is used in an upcoming step.");
+        waitForInputToContinue(scanner);
+
+        String arn = "";
+        try {
+            CompletableFuture<String> future = openSearchActions.describeDomainAsync(domainName);
+            arn = future.join();
+        } catch (RuntimeException rt) {
+            Throwable cause = rt.getCause();
+            if (cause instanceof OpenSearchException openSearchEx) {
```

Comment:
Why do you use "if" here, and use "while" in other places to find the specific exception?
See line 127.

Reply:
fixed